### PR TITLE
feat: storybook for Icon component

### PR DESCRIPTION
**Pull request by rusackas**
### SUMMARY

Adds a storybook entry for the Icon component, and puts that Icon component in a folder.

Sadly, the knobs for the color palette are relying on the `SupersetTheme`... I was having some issues getting `withTheme` to work here, but I'd _rather_ use the `ThemeProvider` to do this. I'll give that another shot later, as it'll be handy in other stories, I'm sure.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

image

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/components/Icon/icon.stories.jsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    Default: null,
+    ...palette,
+  },
+  defaultValue: null,
+};
+
+export const SupersetIcon = () => {
+  return (
+    <>
+      {Object.keys(iconsRegistry).map(iconName => (
```
Comment (etr2460) [inline superset-frontend/src/components/Icon/icon.stories.jsx:50]: A bit of a nit, do you think we could display these icons in such a way that they take up a bit less space? I've liked a design like this lately: https://airbnb.io/lunar/?path=/story/icons--icon-gallery

File: superset-frontend/src/components/Icon/icon.stories.jsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        <>
+          {iconName}: <br />
+          <Icon
```
Comment (etr2460) [inline superset-frontend/src/components/Icon/icon.stories.jsx:53]: Should we add a knob for size as well?

Comment (rusackas) [reply superset-frontend/src/components/Icon/icon.stories.jsx:52]: Hmm... I don't see a `size` prop in use in the codebase, or explicitly supported by the Icon component.

Comment (rusackas) [reply superset-frontend/src/components/Icon/icon.stories.jsx:52]: Otherwise, I'm down to add knobs for _all the things_

Comment (nytai) [reply superset-frontend/src/components/Icon/icon.stories.jsx:52]: width and height are svg props which can be passed into the Icon component, along with viewBox to properly scale to svg.

**Review by nytai — APPROVED** [review]

**Comment by rusackas** [comment]
🏷 svg

File: superset-frontend/src/components/Icon/icon.stories.jsx (hunk 2 above)
Comment (etr2460) [reply superset-frontend/src/components/Icon/icon.stories.jsx:52]: perhaps we should consider abstracting width, height, and viewBox away into a `size` prop? But regardless, that's for a future pr. This is good as is then

**Comment by mistercrunch** [comment]
Oh the label bot is working again!? Came back to life!?!?!!

About the storybook, I'd go for much more density, maybe in a grid? Small text?

**Comment by rusackas** [comment]
Ok, icons are now in a nice little grid... sorry for being a lazy bum :P

And hooray label bot!

**Comment by rusackas** [comment]
I also sorted the icons alphabetically, so it doesn't hurt people's brains.

**Comment by mistercrunch** [comment]
🏷️ storybook

**Comment by rusackas** [comment]
🏷 storybook

**Comment by mistercrunch** [comment]
WHAT? bot only listens to you!?

**Comment by mistercrunch** [comment]
🗑️ 🏷️ storybook

**Comment by rusackas** [comment]
🗑🏷 svg

**Comment by mistercrunch** [comment]
🏷️ .ui

**Comment by rusackas** [comment]
mad panda

**Comment by mistercrunch** [comment]
That's super offensive. Bot revolts against its maker.

**Comment by rusackas** [comment]
I'll go tag some random PR I'm NOT the author of to see if it works there.

**Comment by mistercrunch** [comment]
🏷️ css

**Pull request merged**